maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuilder.java

```diff
@@ -703,15 +671,14 @@ private boolean build( List<ProjectBuildingResult> results, List<MavenProject> p
     }
 
     @SuppressWarnings( "checkstyle:methodlength" )
-    private void initProject( MavenProject project, Map<String, MavenProject> projects,
+    private void initProject( MavenProject project, Map<File, MavenProject> projects,
                               boolean buildParentIfNotExisting, ModelBuildingResult result,
                               Map<File, Boolean> profilesXmls, ProjectBuildingRequest projectBuildingRequest )
     {
         Model model = result.getEffectiveModel();
 
         project.setModel( model );
-        project.setOriginalModel( result.getRawModel() );
-        project.setFile( model.getPomFile() );
```

Comment:
Why is this not required anymore?

Reply:
File was already set on the project when creating it (line 167)
Also, the Model was being abused to hold the file. A model is created with an InputStream, so the file information was added via post-processing. Only the MavenProject is interested in the file, not the Model.